src/runner/system_menu_helper.cpp

```diff
@@ -0,0 +1,127 @@
+#include "pch.h"
+#include "system_menu_helper.h"
+
+#include <interface/powertoy_module_interface.h>
+
+namespace {
+  constexpr int KSeparatorPos = 1;
+  constexpr int KNewItemPos = 2;
+
+  int GenerateItemId() {
+    static int generator = 0xDEADBEEF;
+    return ++generator;
+  }
+}
+
+SystemMenuHelper& SystemMenuHelperInstace() {
+  static SystemMenuHelper instance;
+  return instance;
+}
+
+void SystemMenuHelper::SetConfiguration(PowertoyModuleIface* module, std::vector<ItemInfo>& config) {
+  Reset(module);
+  Configurations[module] = config;
+  for (auto& [window, modules] : ProcessedModules) {
+    // Unregister module. After system menu is opened again, new configuration will be applied.
+    modules.erase(std::remove(std::begin(modules), std::end(modules), module), std::end(modules));
+  }
+}
+
+void SystemMenuHelper::ProcessSelectedItem(PowertoyModuleIface* module, HWND window, const wchar_t* itemName) {
+  for (auto& item : Configurations[module]) {
+    if (!wcscmp(itemName, item.name.c_str()) && item.checkBox) {
+      // Handle check/uncheck action only if specified by module configuration.
+      for (auto& [id, data] : IdMappings) {
+        if (data.second == itemName) {
+          HMENU systemMenu = GetSystemMenu(window, false);
+          int state = (GetMenuState(systemMenu, id, MF_BYCOMMAND) == MF_CHECKED) ? MF_UNCHECKED : MF_CHECKED;
+          CheckMenuItem(systemMenu, id, MF_BYCOMMAND | state);
+          break;
+        }
+      }
+      break;
+    }
+  }
+}
+
+bool SystemMenuHelper::Customize(PowertoyModuleIface* module, HWND window) {
+  for (const auto& m : ProcessedModules[window]) {
+    if (module == m) {
+      return false;
+    }
+  }
+  AddSeparator(module, window);
+  for (const auto& info : Configurations[module]) {
+    AddItem(module, window, info.name, info.enable);
+  }
+  ProcessedModules[window].push_back(module);
+  return true;
+}
+
+void SystemMenuHelper::Reset(PowertoyModuleIface* module) {
+  for (auto& [window, modules] : ProcessedModules) {
+    for (auto& [id, data] : IdMappings) {
+      HMENU sysMenu{ nullptr };
+      if (data.first == module && (sysMenu = GetSystemMenu(window, false))) {
```

Comment:
the `sysMenu = GetSystemMenu(window, false)` check should to the upper level for, no need to check all IdMappings if this is false.